### PR TITLE
task 3 review by id completed

### PR DESCRIPTION
**Pull request by roseannac21**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -30,4 +30,14 @@ describe("app tests", () => {
...
+        expect.objectContaining({review_id: expect(5), title: expect('Proident tempor et.'), category: expect('social deduction'), designer: expect('Seymour Buttz'), owner: expect('mallionaire'), review_body: expect('Labore occaecat sunt qui commodo anim anim aliqua adipisicing aliquip fugiat. Ad in ipsum incididunt esse amet deserunt aliqua exercitation occaecat nostrud irure labore ipsum. Culpa tempor non voluptate reprehenderit deserunt pariatur cupidatat aliqua adipisicing. Nostrud labore dolor fugiat sint consequat excepteur dolore irure eu. Anim ex adipisicing magna deserunt enim fugiat do nulla officia sint. Ex tempor ut aliquip exercitation eiusmod. Excepteur deserunt officia voluptate sunt aliqua esse deserunt velit. In id non proident veniam ipsum id in consequat duis ipsum et incididunt. Qui cupidatat ea deserunt magna proident nisi nulla eiusmod aliquip magna deserunt fugiat fugiat incididunt. Laboris nisi velit mollit ullamco deserunt eiusmod deserunt ea dolore veniam.'), review_img_url: expect('https://images.pexels.com/photos/209728/pexels-photo-209728.jpeg?w=700&h=700'), created_at: expect('2021-01-07T09:06:08.077Z'), votes: expect(5)})
+        })
```
Comment (HeatherEver) [inline __tests__/app.test.js:67]: Please can we add some error handling tests for this endpoint? 
think of paths and errors such as `/api/reviews/9999` and `/api/reviews/wrong`

**Review by HeatherEver — CHANGES_REQUESTED** [review]
Nice work again here Roseanna but you need to be adding some tests for error handling now and think about how they are going to be handled - once error handling has been added can you re-submit this pull request? 😊

File: __tests__/app.test.js (hunk 1 above)
Comment (roseannac21) [reply __tests__/app.test.js:67]: okay will do this now :)

File: db/model.js [hunk 2]
```diff
@@ -16,4 +16,22 @@ const fetchReviews = () => {
...
 }
 
-module.exports = { fetchCategories, fetchReviews }
\ No newline at end of file
+const fetchReviewById = (id) => {
+    const parsedId = parseInt(id)
+    
+    if (id > 24) {
+        return Promise.reject({status: 404, msg: "invalid review ID"})
+    };
```
Comment (HeatherEver) [inline db/model.js]: This works for now but what if later we are able to add reviews to the database this id will go higher than 24 and therefore trigger this error - is there another way we can check if the id exists? If it doesn't exist, we won't receive anything back in our rows array.

File: db/model.js [hunk 3]
```diff
@@ -16,4 +16,22 @@ const fetchReviews = () => {
...
+
+    if (isNaN(parsedId) === true) {
+        return Promise.reject({status: 400, msg: "invalid data type"})
+    }
```
Comment (HeatherEver) [inline db/model.js]: With this one is there a way we could rely on this error going to our controller, passing it to our catch block using `next` and then pass it to our error handlers in app. Try console logging this error within your error handling functions in app and see if it has a psql error code you can use to then return the message you want.

**Review by HeatherEver — CHANGES_REQUESTED** [review]
Hey Roseanna, nice work but we could do with making your error handling a little more dynamic and passing your 400 errors to error handlers in your app. Think about how you can do this and use console logs to see what these errors look like 😊

**Pull request closed without merging**